src/tpg_udp.c

```diff
@@ -379,16 +381,16 @@ static struct udp_hdr *udp_build_hdr(udp_control_block_t *ucb,
     udp_hdr->dst_port = rte_cpu_to_be_16(ucb->ucb_l4.l4cb_dst_port);
 
     udp_hdr->dgram_len = rte_cpu_to_be_16(dgram_len + udp_hdr_len);
+    mbuf->l4_len = udp_hdr_len;
+    ip_hdr_len = ((ipv4_hdr->version_ihl & 0x0F) << 2);
```

Comment:
No real need to compute `ip_hdr_len` here. I would keep it inside the `if` below as it's only used there.